gen/temp.*.go not cleaned up sometimes on Windows

I recently updated to v3.0.6 on some projects and started seeing this file under gen.

It was introduced with https://github.com/goadesign/goa/pull/2281

It gets left behind when I run both a gen and example in a single command like:

`goa example goa.design/examples/basic/design && goa gen goa.design/examples/basic/design`

If I run the two command independently, it doesn't happen. I can sometimes get it if I call these two from make targets without the && where the commands end up executing back to back.

It happens on a windows development box. Haven't seen the behavior on any of my linux build nodes. I suspect some kind of os level file locking issue and some kind of race condition between the two invocations using the same filename. Seems far fetched but I can't think of anything else.

I see similar with the goa* (goa34252433) directory in the pwd sometimes on windows but not sure if it is related and those do not break my builds.

There is no feedback if the `defer os.Remove(dummy.Name())` fails. Perhaps changing that to something like:

```
defer func() { if err := os.Remove(dummy.Name()); if err != nil {
// panic ? report the error on os.Stderr?
}
```

would at least give me some feedback that it happened and why.

I've been able to ignore the ./goa* issue because it doesn't affect my build, but the gen/temp.*.go files cause my builds to fail when I lint with goimports

It isn't an option for me to ditch my windows development environment ...insert sad face emoji... so another solution for others stuck in my predicament might be to keep the gen file around maybe named consistently something like doc.go with a package level comment and formatted so it passes goimports linting checks.

Right now, I'm working around this in my makefile with:

`if stat -t  gen/temp.*.go >/dev/null 2>&1; then rm gen/temp.*.go; fi`

after I run gen and example

I was able to reproduce the behavior outside of goa with a trivial program. Adding a dummy.Close() when I was done with the file seems to have resolved the issue:
```
package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"path/filepath"
)

func main() {
	base, err := os.Getwd()
	if err != nil {
		panic(err)
	}
	path := filepath.Join(base, "gen")
	if err := os.MkdirAll(path, 0777); err != nil {
		panic(err)
	}

	dummy, err := ioutil.TempFile(path, "temp.*.go")
	if err != nil {
		panic(err)
	}
	if _, err := fmt.Fprintf(os.Stderr, "opened %s\n", dummy.Name()); err != nil {
		panic(err)
	}
	defer func() {
		if err := os.Remove(dummy.Name()); err != nil {
			panic(err)
		}
		if _, err := fmt.Fprintf(os.Stderr, "removed %s\n", dummy.Name()); err != nil {
			panic(err)
		}
	}()
	if err = ioutil.WriteFile(dummy.Name(), []byte("package gen"), 0644); err != nil {
		panic(err)
	}
	if err = dummy.Close(); err != nil {
		panic(err)
	}
}
```

Without the close, I get this every time:
```
$ /c/dev/go-experiments/GOPATH/bin/temp-files.exe
opened C:\...\Temp\gen\temp.476452543.go
panic: remove C:\..\Temp\gen\temp.476452543.go: The process cannot access the file because it is being used by another process.

goroutine 1 [running]:
main.main.func1(0xc000094018)
        C:/dev/go-experiments/temp-files/tempfile.go:29 +0x13a
main.main()
        C:/dev/go-experiments/temp-files/tempfile.go:38 +0x2d5
```
once I added the close call, I get this every time:
```
$ /c/dev/go-experiments/GOPATH/bin/temp-files.exe
opened C:\...\Temp\gen\temp.824370235.go
removed C:\...\Temp\gen\temp.824370235.go

```

--- a/codegen/generator/generate.go
+++ b/codegen/generator/generate.go
@@ -12,7 +12,7 @@ import (
 )
 
 // Generate runs the code generation algorithms.
-func Generate(dir, cmd string) ([]string, error) {
+func Generate(dir, cmd string) (outputs []string, err1 error) {
 	// 1. Compute design roots.
 	var roots []eval.Root
 	{
@@ -40,8 +40,16 @@ func Generate(dir, cmd string) ([]string, error) {
 		if err != nil {
 			return nil, err
 		}
-		defer os.Remove(dummy.Name())
-		if err = ioutil.WriteFile(dummy.Name(), []byte("package gen"), 0644); err != nil {
+		defer func() {
+			if err := os.Remove(dummy.Name()); err != nil {
+				outputs = nil
+				err1 = err
+			}
+		}()
+		if _, err = dummy.Write([]byte("package gen")); err != nil {
+			return nil, err
+		}
+		if err = dummy.Close(); err != nil {
 			return nil, err
 		}
 
@@ -97,7 +105,6 @@ func Generate(dir, cmd string) ([]string, error) {
 	}
 
 	// 8. Compute all output filenames.
-	var outputs []string
 	{
 		outputs = make([]string, len(written))
 		cwd, err := os.Getwd()

